### PR TITLE
WIP: TextComponent with sass mixin & variables implemented.

### PR DESCRIPTION
**Pull request by Ronan-Codes**

### CODE REVIEW

File: src/components/TextComponent/TextComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    error?:boolean,
+    onChange?:Function
+    // onChange?:React.ChangeEvent<HTMLInputElement>,
+    // All other props (...rest). Is this right???
+    // [x:string]: any;
+}
+
+const TextComponent: FC<TextInputProps> = ({
+    inputValue = "",
+    placeholderText = "",
```
Comment (GoldenaArcher) [inline src/components/TextComponent/TextComponent.tsx:19]: I think `placeholder` can be omitted, if you need to pass placeholder, it'll be `placehonder: "some text"` which will be handled by `...rest`

File: src/components/TextComponent/TextComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  return (
+    <div className="text__component">
+        <input
+          placeholder={isTherePlaceholder()}
+          className={classList()}
+          type="text"
+          value={input}
+          onChange={handleInputChange}
+          {...rest}
+        />
```
Comment (GoldenaArcher) [inline src/components/TextComponent/TextComponent.tsx:62]: To ensure `type` is `text`, you can add `type=text` at bottom

File: src/styles/_mixin.scss [hunk 3]
```diff
@@ -74,3 +74,26 @@
   border-radius: $border-radius;
   border-color: $border-color;
 }
+
+// TextComponent Addition
+@mixin input-color($background-color, $border-width, $border-color) {
```
Comment (GoldenaArcher) [inline src/styles/_mixin.scss:79]: Redundant code here, you can reuse something here including CSS cascading rule.

File: src/components/TextComponent/TextComponent.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    const thisInputChange = (e: React.ChangeEvent<HTMLInputElement>)=> {
+        setthisInput(e.target.value)
+     }
+
+    return(
+        <>
+            <TextComponent  inputValue='Input' error={false}/>
+            <br></br>
+            <TextComponent inputValue={thisInput} placeholderText="Placeholder" error={true} onChange={thisInputChange}/>
```
Comment (GoldenaArcher) [inline src/components/TextComponent/TextComponent.stories.tsx:21]: Updating corresponding to the comment I have below

**Review by GoldenaArcher — CHANGES_REQUESTED** [review]
Check comments above

File: src/styles/_mixin.scss (hunk 3 above)
Comment (Ronan-Codes) [reply src/styles/_mixin.scss:79]: Oh yeah I meant to delete input-color, because input-colors below it replaces it

**Pull request closed without merging**